src/preferenceCenter/preferenceCenter.commands.js

```diff
@@ -0,0 +1,183 @@
+      const attributes = preferenceCenter.getAttributes();
+      const decId = attributes['data-param-dec-id'];
+      // If it's not a token -> convert to token
+      if (decId >= 0) {
+        const dynConId = PreferenceCenterCommands.getPcStoreId(attributes['data-param-dec-id']);
+
+        const dynConTarget = mQuery(dynConId);
+        const dynConName = dynConTarget.find(`${dynConId}_tokenName`).val();
+        const dynConToken = `{dynamiccontent="${dynConName}"}`;
+
+        // Clear id because it's reloaded by Mautic and this prevent slot to be destroyed by GrapesJs destroy event on close.
+        // preferenceCenter.addAttributes({ 'data-param-dec-id': '' });
+        preferenceCenter.set('content', dynConToken);
+      }
+    });
+
+    return preferenceCenter.length;
+  }
+
+  /**
+   * Build and display the Dynamic Content editor popup/modal window.
+   * @param {Model} component The current grapesjs component
+   */
+  // eslint-disable-next-line class-methods-use-this
+  showPreferenceCenterEditor(editor, options) {
+    this.dcPopup = PreferenceCenterCommands.buildPreferenceCenterPopup();
+
+    this.addPreferenceCenterEditor(editor, options);
+
+    const title = Mautic.translate('grapesjsbuilder.preferenceCenterBlockLabel');
+    const modal = editor.Modal;
+    modal.setTitle(title);
+
+    editor.Modal.setContent(this.dcPopup);
+    modal.open();
+    modal.onceClose(() => editor.stopCommand('preset-mautic:dynamic-content-open'));
+  }
+
+  /**
+   * Build the basic popup/modal frame to hold the dynamic content editor
+   * @returns HTMLDivElement
+   */
+  static buildPreferenceCenterPopup() {
+    const content = document.createElement('div');
+    content.setAttribute('id', 'dynamic-content-popup');
+
+    const codePopup = document.createElement('div');
+    codePopup.appendChild(content);
+
+    return codePopup;
+  }
+
+  /**
+   * Load Dynamic Content editor and append to the codePopup Modal
+   */
+  addPreferenceCenterEditor(editor, options) {
+    const { target } = options;
+    const dcComponent = target || editor.getSelected();
+    if (!dcComponent) {
+      throw new Error('No dc components found');
+    }
+
+    const attributes = dcComponent.getAttributes();
+    // const popupContent = dcPopup.querySelector('#dynamic-content-popup');
+
+    // do we need both?
+    // console.warn({ dcPopup });
+    // console.warn({ popupContent });
+
+    // get the dynamic content editor
+    const focusForm = mQuery(PreferenceCenterCommands.getPcStoreId(attributes['data-param-dec-id']));
+    if (focusForm.length <= 0) {
+      throw new Error(
+        `No preferenceCenter email form found for '${attributes['data-param-dec-id']}'`
+      );
+    }
+
+    // Show if hidden
+    focusForm.removeClass('fade');
+    // Hide delete default button
+    focusForm.find('.tab-pane:first').find('.remove-item').hide();
+
+    // Insert inside popup
+    mQuery(this.dcPopup).empty().append(focusForm.detach());
```

Comment:
"dc" stands for dynamic content. please change all occurrences to "prefereceCenter" or "pc"

Reply:
same for "dynCon"